util/generate-ami-list.py

```diff
@@ -370,22 +353,32 @@ def update_amis_txt(amis_txt_file, amis):
             if credential_tuple.strip()
         ]
 
-    if (args.version and args.date) or (args.cookbook_git_ref and args.node_git_ref):
-        regions = get_all_aws_regions_from_ec2(region)
-        amis_dict = get_ami_list_from_ec2(
+    if args.version and args.date:
```

Comment:
nit: I don't think this logic is used anywhere in Jenkins? I don't think we ever call this script with the version. Can we double check why this is needed?

Ideally it might also help to add comment for when each condition is used in Jenkins and what we are trying to achieve, so we don't have to look at Jenkins script.

For example, we would be using `elif args.cookbook_git_ref and args.node_git_ref` logic in build_and_test to enforce that AMIs needs to be created with the same git ref(and the same build_date actually, since build_date is provided in the call). This might not be immediately clear when we look at this script or the Jenkins file alone.

Reply:
This is a very good idea - I found myself doing this a lot while working on it. I'll add another commit with this change.